api/example.com/resource/gpu/v1alpha1/api_test.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2024 The Kubernetes Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGpuConfigNormalize(t *testing.T) {
+	tests := []struct {
```

Comment:
nit: Making it a map would change an order of test cases for each run, which can help ensuring that they don't influence each other.
```suggestion
	tests := map[string]struct {
```